<launch>

    <arg name="launch_state" default="sim"/> 

    <arg name="scan_topic" default="/scan_modified" />
    <arg name="imu_topic"  default="/imu/data" />
    <arg name="odom_topic" default="/odom" />
    <arg name="point_cloud_topic" default="/cv/lane_detections_viz" />

    <arg name="base_frame"  default="base_link"/>
    <arg name="odom_frame"  default="odom"/>

    <node name="cartographer_node" pkg="cartographer_ros" type="cartographer_node" args="-configuration_directory $(find description)/config -configuration_basename caffeine.lua" output="screen">

        <remap from="imu" to="$(arg imu_topic)" />
        <remap from="odom" to="$(arg odom_topic)" />
        <remap from="scan" to="$(arg scan_topic)" />
        <!-- <remap from="points2" to="$(arg point_cloud_topic)" /> -->
        <remap from="fix" to="/gps/fix"/>
    </node>

    <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros" type="cartographer_occupancy_grid_node" args="-resolution 0.05" />


</launch>
